File: instrumentos_api/src/main/java/org/example/entities/Instrumento.java
package org.example.entities;

import jakarta.persistence.*;
import lombok.*;

@Entity
@AllArgsConstructor
@Getter
@Setter
@NoArgsConstructor
@Table(name="instrumento")
@Builder
public class Instrumento {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String instrumento;
    private String marca;
    private String modelo;
    private String imagen;
    private double precio;
    private String costoEnvio;
    private int cantidadVendida;
    
    @Column(length = 2048)
    private String descripcion;

    @ManyToOne
    @JoinColumn(name = "id_categoria", nullable = false)
    private CategoriaInstrumento categoria;

}

File: src/types/Instrumento.ts
export interface Instrumento {
  id: number
  instrumento: string
  marca: string
  modelo: string
  imagen: string
  precio: number
  costoEnvio: string
  cantidadVendida: number
  descripcion: string
  categoria: string
}
export type FormField = {
  label: string
  name: string
  placeholder?: string
  type: string
  value?: string | number
}
